server/async-worker/index.js

```diff
@@ -0,0 +1,65 @@
+const { Op } = require("sequelize");
+const callScraper = require("../scraper");
+const { Mention, User } = require("../models");
+
+// A rough draft of how new mentions would get pulled into the db
+// without using an explicit /POST route as well as using
+// a given user's credentials.
+module.exports = async function asyncWorker() {
+  const SECONDS = 1;
+  const MILLISECONDS = 1000;
+  // TODO refactor this into smaller functions
+  console.log("[Scraper] Calling async scraper");
+
+  const users = await User.findAll();
+  for (let user of users) {
+    let mentions = await callScraper(user.company);
+    let count = 0;
+
+    for (let m of mentions) {
+      let col, val, timeConvention;
+      // if this is a twitter mention
+      if (m.platform == "Twitter") {
+        col = "content";
+        val = m.content;
+        timeConvention = SECONDS;
+      }
+      // if this is a reddit mention
+      else if (m.platform == "Reddit") {
+        col = "title";
+        val = m.title;
+        timeConvention = MILLISECONDS;
+      }
```

Comment:
I think we want to get rid of this `if-else`, so we need to discuss how to deal with the problem that tweets don't have a title while posts on Reddit have.

Reply:
However with the data returned from the reddit scraper, while each thread will have a title returned not every thread will have the content field populated, which is the opposite of the twitter scraper. The idea I was trying to get at here is to  select a field from the obtained mentions that serve as a unique identifier for the mentions. In this case it would be the title for reddit(and possibly other platforms), and since twitter has an empty title, the content field would be the identifying value when querying the db.

Reply:
The mentions we get from Reddit will definitley have non-empty content, since as Aecio indicated, I have filtered out all empty mentions. For tweets, like I said, let's consult Aecio about how to deal with the lack of title first, since it will be a problem when displaying in the main page. so we need to solve it anyway. The reason I think we need to get rid of this `if-else` is, image we have 10+ platforms, then this part of code will be tedious.